community/modules/compute/schedmd-slurm-gcp-v5-partition/variables.tf

```diff
@@ -0,0 +1,298 @@
+  type        = bool
+  default     = false
+}
+
+variable "disable_smt" {
+  type        = bool
+  description = "Disables Simultaneous Multi-Threading (SMT) on instance."
+  default     = false
+}
+
+variable "labels" {
+  description = "Labels to add to partition compute instances. List of key key, value pairs."
+  type        = any
+  default     = {}
+}
+
+variable "min_cpu_platform" {
+  description = "The name of the minimum CPU platform that you want the instance to use."
+  type        = string
+  default     = null
+}
+
+variable "on_host_maintenance" {
+  type        = string
+  description = "Instance availability Policy"
+  default     = "MIGRATE"
+}
+
+variable "gpu" {
+  description = "Definition of requested GPU resources"
+  type = object({
+    count = number,
+    type  = string
+  })
+  default = null
+}
+
+variable "network_storage" {
+  description = "An array of network attached storage mounts to be configured on the partition compute nodes."
+  type = list(object({
+    server_ip     = string,
+    remote_mount  = string,
+    local_mount   = string,
+    fs_type       = string,
+    mount_options = string
+  }))
+  default = []
+}
+
+variable "preemptible" {
+  description = "Should use preemptibles to burst"
+  type        = string
+  default     = false
+}
+
+variable "service_account" {
+  type = object({
+    email  = string
+    scopes = set(string)
+  })
+  description = <<-EOD
+    Service account to attach to the instances. See
+    'main.tf:local.service_account' for the default.
+    EOD
+  default = {
+    email = "default"
+    scopes = [
+      "https://www.googleapis.com/auth/cloud-platform",
+    ]
+  }
+}
+
+variable "shielded_instance_config" {
+  type = object({
+    enable_integrity_monitoring = bool
+    enable_secure_boot          = bool
+    enable_vtpm                 = bool
+  })
+  description = <<-EOD
+    Shielded VM configuration for the instance. Note: not used unless
+    enable_shielded_vm is 'true'.
+    * enable_integrity_monitoring : Compare the most recent boot measurements to the
+      integrity policy baseline and return a pair of pass/fail results depending on
+      whether they match or not.
+    * enable_secure_boot : Verify the digital signature of all boot components, and
+      halt the boot process if signature verification fails.
+    * enable_vtpm : Use a virtualized trusted platform module, which is a
+      specialized computer chip you can use to encrypt objects like keys and
+      certificates.
+    EOD
+  default = {
+    enable_integrity_monitoring = true
+    enable_secure_boot          = true
+    enable_vtpm                 = true
+  }
+}
+
+variable "subnetwork_self_link" {
+  type        = string
+  description = "Subnet to deploy to. Only one of network or subnetwork should be specified."
+  default     = ""
+}
+
+variable "exclusive" {
+  description = "Exclusive job access to nodes"
+  type        = bool
+  default     = false
```

Comment:
I am not sure how exclusive and placement interact on the Slurm side, but as discussed previously, ideally exclusive would follow the setting for placement unless they are explicitly set to be different. My recommendation would be to have the default be null, and set it to follow placement in locals (if not explicitly set). At a minimum we should document when this is being ignored. 

The default doesn't really make sense at the moment (placement on, exclusive off)

Reply:
Exclusive is set based on placement: https://github.com/SchedMD/slurm-gcp/blob/master/terraform/slurm_cluster/modules/slurm_partition/main.tf#L29

So this works, but is not intuitive. I'll update the default to true.

Reply:
nit: it might be worth 1 line in the description to say, "exclusive will always be enabled when placement is enabled".